src/Game/AI.cpp

```diff
@@ -338,6 +361,24 @@ namespace ai{
         return ret;
     }
 
+    bool teamHasQuaffle(const std::shared_ptr<const gameModel::Environment> &env, const std::shared_ptr<gameModel::Player> &player) {
+        std::shared_ptr<gameModel::Team> team = env->team1;
```

Comment:
getTeam
```suggestion
        auto team = env->getTeam(player);
```

Reply:
kommt man nicht auch evtl ohne die team hilfsvariable aus?

Reply:
man könnte ja auch sich zuerst über die Position vom Ball schauen ob überhaupt ein Spieler drauf steht...
```suggestion
        auto optionalPlayer = env->getPlayer(env->quaffle->position);
        if(optionalPlayer.has_value()){
        if(env->getTeam(optionalPlayer)->side == env->getTeam(player)){
        return true;
        }
        }
```

Reply:
ist die Frage ob es so viel schöner wird?

Reply:
getTeam scheint mit optional Player nicht zu funktionieren

Reply:
Ja, das muss dann ja auch `env->getTeam(optionalPlayer.value())->side` sein